### DISCUSSION TITLE
Incorporating the Node.js into a PHP Infrastructure

### DISCUSSION THREAD

**Discussion by MillennialInvestments** [knicola/tdameritradejs#43] [Q&A]
Hello and thank you for taking the time to discuss this with me.

We are building an investment and accounting analytical platform for financial investors that we have constructed via HTML5/CSS3/Javascript/PHP 7.4/MySQL. Our mission is to utilize this TDA API Node.js Library to allow our users to simply automatically pull all of their transactions/open positions from their TDA Ameritrade account so that our software platform can log and provide analytical information for better investment decisions and helping investors learn how to make better financial decisions when determining the next best trade for them.

I was wondering if you knew of a way that this Node.js project could be incorporated in an infrastructure such as ours with seamless and easy-to-use integration to accomplish our mission.

**Comment by knicola** [comment]
Hello,

You could write an internal API service in nodejs and call it from PHP with something like guzzle. Though depending on what exactly you need from TD it might be easier to write a limited scope API client in PHP instead.
